molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java

```
@@ -391,6 +408,210 @@ public String viewAttributeMapping(@RequestParam(required = true) String mapping
 		return VIEW_ATTRIBUTE_MAPPING;
 	}
 
+	@RequestMapping(value = "/attributemappingfeedback")
+	public String attributeMappingFeedback(@RequestParam(required = true) String mappingProjectId,
+			@RequestParam(required = true) String target, @RequestParam(required = true) String source,
+			@RequestParam(required = true) String targetAttribute, @RequestParam(required = true) String algorithm,
+			Model model)
+	{
+		MappingProject project = mappingService.getMappingProject(mappingProjectId);
+
+		MappingTarget mappingTarget = project.getMappingTarget(target);
+		EntityMapping entityMapping = mappingTarget.getMappingForSource(source);
+
+		AttributeMapping algorithmTest;
+
+		if (entityMapping.getAttributeMapping(targetAttribute) == null)
+		{
+			algorithmTest = entityMapping.addAttributeMapping(targetAttribute);
+			algorithmTest.setAlgorithm(algorithm);
+		}
+		else
+		{
+			algorithmTest = entityMapping.getAttributeMapping(targetAttribute);
+			algorithmTest.setAlgorithm(algorithm);
+		}
+
+		try
+		{
+			Collection<String> sourceAttributeNames = algorithmService.getSourceAttributeNames(algorithm);
+			if (!sourceAttributeNames.isEmpty())
+			{
+				model.addAttribute("sourceAttributeNames", sourceAttributeNames);
+			}
+		}
+		catch (Exception e)
+		{
+
```

Do not silently ignore Exception.